src/signed_fetch.rs

```diff
@@ -0,0 +1,393 @@
+// This was built following the https://adr.decentraland.org/adr/ADR-44
+use dcl_crypto::{Address, AuthChain, AuthLink, Authenticator, Web3Transport};
+use std::{
+    collections::HashMap,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+const AUTH_CHAIN_HEADER_PREFIX: &str = "x-identity-auth-chain-";
+const AUTH_TIMESTAMP_HEADER: &str = "x-identity-timestamp";
+const AUTH_METADATA_HEADER: &str = "x-identity-metadata";
```

Comment:
Are the headers case-sensitive? In that case, shouldn't we normalize them before validations?